cpp/cmake/thirdparty/CUDF_GetArrow.cmake

```diff
@@ -14,11 +14,12 @@
 # limitations under the License.
 #=============================================================================
 
-function(find_and_configure_arrow VERSION BUILD_STATIC)
+function(find_and_configure_arrow VERSION BUILD_STATIC ENABLE_S3)
 
     set(ARROW_BUILD_SHARED ON)
     set(ARROW_BUILD_STATIC OFF)
     set(CPMAddOrFindPackage CPMFindPackage)
+    set(ARROW_BUILD_S3 OFF)
```

Comment:
Nitpick, but can you move this a line above `CPMAddOrFindPackage` so it is with the other `ARROW_BUILD` variables.